examples/conanfile/package_id/settings_and_options.rst

```diff
@@ -0,0 +1,202 @@
+.. _examples_conanfile_package_id_settings_options:
+
+How settings and options of a recipe influence its pacakge ID
+=============================================================
+
+In Conan, a package ID is a unique identifier for a package that takes into account all the factors that affect its contents.
+These factors include recipe options and settings as well as requirements or tool requirements.
+
+Let's see how settings and options affect the package ID and some examples where they should not.
+
+How settings influence the package ID
+-------------------------------------
+
+Settings are development project-wide variables, like the compiler, its version, or the OS itself.
+These variable values have to be defined, they should match the values of our development environment, and they cannot have a default value.
+
+For example, let's define a recipe that generates packages that are only OS dependent:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os"  # Only OS setting affects the package ID
+
+If we generate a package from this recipe for Linux we will get the following package ID:
+
+.. code-block:: bash
```

Comment:
I think it would be superuseful to highlight the package id's lines in all the code blocks

Reply:
done